src/Control/Distributed/Spark/SQL/DataFrame.hs

```diff
@@ -32,3 +33,168 @@ debugDF df = call df "show" []
 
 join :: DataFrame -> DataFrame -> IO DataFrame
 join d1 d2 = call d1 "join" [coerce d2]
+
+newtype DataFrameReader =
+    DataFrameReader (J ('Class "org.apache.spark.sql.DataFrameReader"))
+instance Coercible DataFrameReader
+                   ('Class "org.apache.spark.sql.DataFrameReader")
+
+newtype DataFrameWriter =
+    DataFrameWriter (J ('Class "org.apache.spark.sql.DataFrameWriter"))
+instance Coercible DataFrameWriter
+                   ('Class "org.apache.spark.sql.DataFrameWriter")
+
+read :: SQLContext -> IO DataFrameReader
+read sc = call sc "read" []
+
+write :: DataFrame -> IO DataFrameWriter
+write df = call df "write" []
+
+readParquet :: [Text] -> DataFrameReader -> IO DataFrame
+readParquet fps dfr = do
+    jfps <- reflect fps
+    call dfr "parquet" [coerce jfps]
+
+writeParquet :: Text -> DataFrameWriter -> IO ()
+writeParquet fp dfw = do
+    jfp <- reflect fp
+    call dfw "parquet" [coerce jfp]
+
+newtype StructType =
+    StructType (J ('Class "org.apache.spark.sql.types.StructType"))
+instance Coercible StructType
+                   ('Class "org.apache.spark.sql.types.StructType")
+
+type JStructFieldClass = 'Class "org.apache.spark.sql.types.StructField"
```

Comment:
Do away with this type synonym, for consistency with the rest of the bindings and because making a type in the class name anywhere else will be caught statically anyways.